File: tests/eval_c/352.c
#include <assert.h>
#include <string.h>

void f352(const char *s) {
    char a = '\0', b = '\0', x = '\0';
    
    if (strlen(s) == 2) {
        a = s[0];
        b = s[1];
    }
    
    if (strlen(s) == 3) {
        a = s[0];
        x = s[1];
        b = s[2];
    }
    
    assert(a == 'a');
    assert(b == 'b');
}

File: tests/oopsla25/352.py
#!/usr/bin/env python3

import sys

def f352(s: str):
  a = ""
  b = ""
  if len(s) == 2:
    a,b = s
  if len(s) == 3:
    a,x,b = s
  assert a == "a"
  assert b == "b"
   
if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f352(my_string)
